src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const ACTION_CLEAR = 'clear';
+  const ACTION_ADD_PROPERTIES = 'addProperties';
+  const ACTION_REMOVE__PROPERTIES = 'removeProperties';
+  const transformStateResult = [];
+  const stateClone = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case ACTION_CLEAR:
+        const keys = Object.keys(stateClone);
+
+        for (const key of keys) {
+          delete stateClone[key];
+        }
```

Comment:
Instead of iterating over keys and deleting each one, you can simply assign `stateClone` to an empty object to clear it, which is more performant and concise.